tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SymbStateRewriterAutoWithArrays.scala

```diff
@@ -2,4 +2,6 @@ package at.forsyte.apalache.tla.bmcmt
 
 import at.forsyte.apalache.tla.bmcmt.smt.SolverContext
 
-class SymbStateRewriterAutoWithArrays(_solverContext: SolverContext) extends SymbStateRewriterAuto(_solverContext) {}
+class SymbStateRewriterAutoWithArrays(_solverContext: SolverContext) extends SymbStateRewriterAuto(_solverContext) {
+  override protected val impl = new SymbStateRewriterImplWithArrays(solverContext, exprGradeStore)
+}
```

Comment:
I'm not sure if we strictly need it, I simply saw that it uses a `SymbStateRewriter` and I extended it. It might break `create` in `RewriterBase` if there is a mismatch between the state rewriter and the solver context.